From 64eaf9433dd0e689ea84f10d49ebf438d7765dff Mon Sep 17 00:00:00 2001
From: P Praneesh <quic_ppranees@quicinc.com>
Date: Mon, 3 Jan 2022 12:03:07 +0530
Subject: [PATCH 04/14] ath12k: fix qmi server arrive callback by correcting
 mhi channel configs

Channel execution environment mask expected by mhi driver differs from host
mhi config causes event stall after qmi init. Fix ee_mask value in mhi channel config.

Signed-off-by: P Praneesh <quic_ppranees@quicinc.com>
---
 drivers/net/wireless/ath/ath12k/mhi.c | 10 ++++++----
 1 file changed, 6 insertions(+), 4 deletions(-)

--- a/drivers/net/wireless/ath/ath12k/mhi.c
+++ b/drivers/net/wireless/ath/ath12k/mhi.c
@@ -59,6 +59,7 @@ static struct mhi_channel_config ath12k_
 		.offload_channel = false,
 		.doorbell_mode_switch = false,
 		.auto_queue = false,
+		.auto_start = true,
 	},
 	{
 		.num = 21,
@@ -73,6 +74,7 @@ static struct mhi_channel_config ath12k_
 		.offload_channel = false,
 		.doorbell_mode_switch = false,
 		.auto_queue = true,
+		.auto_start = true,
 	},
 };
 
